jwt/jwt.go

```diff
@@ -124,7 +127,17 @@ func (js jwtSource) Token() (*oauth2.Token, error) {
 	}
 	h := *defaultHeader
 	h.KeyID = js.conf.PrivateKeyID
-	payload, err := jws.Encode(&h, claimSet, pk)
+	payload := ""
+	var err error = nil
+	if js.conf.SignerProvider == nil {
+		var pk *rsa.PrivateKey
+		pk, err = internal.ParseKey(js.conf.PrivateKey)
```

Comment:
Looks like you forgot to return if there's an error here.

```suggestion
		pk, err = internal.ParseKey(js.conf.PrivateKey)
		if err != nil {
			return nil, err
		}
```